dedupe-API.md

```diff
@@ -0,0 +1,30 @@
+
+
+# API
+
+## `/match`
+
+Returns the CJ identifier that the dedupe service believes matches a HMIS identifier (if any).
+
+### Parameters
+- `hmis`: a valid hmis id
+
+### Response
+
+- `404` Empty json object, If the value of `hmis` is not a id known to this service
+- `204` Empty json object, The service can not find a CJ identifier that is a possible match
+- `200` `{"cj": "best matching cj id", "score": score_of_that_match_(float)"}`
+
+### Example
+
+```bash
+curl http:/service.url/match?hmis=10
+{"cj": "ad8aa", "score": .89}
+```
+
+## `/crosswalk`
+
+Returns the current crosswalk table as CSV
+
+# Updating
+The crosswalk will be updated daily, after the HMIS updates
```

Comment:
How significantly would this change if we were to have three or four data sources instead of two? (Thinking about our 6 month plan.)

Reply:
I thought this was a pilot phase? For a production project, we would recommend not doing this batch matching but having an on-demand matching API.

Reply:
It is a pilot! Though I am starting to document resource needs for beyond pilot and what supporting a production project looks like. (ie, knowing that we'll want to switch to on-demand matching).

Reply:
Great! Let's keep this as the API for the pilot phase, and I'll supply other docs for the production API.